frontend/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -91,7 +92,7 @@ function renderFigure({
   const [spec, setSpec] = useState(generateSpec())
 
   // Update config and spec references iff the theme or props change
-  useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
since `useLayoutEffect` is unusual, consider adding a comment about why we use it instead of `useEffect`, or include a link to an explanation

Reply:
Sounds good.